xc/device.ml

```diff
@@ -1475,19 +1469,16 @@ module Vusb = struct
     debug "qom list to check if usb device is attached to vm";
     let is_running = Qemu.is_running ~xs domid in
     match is_running with
-    | true  -> (
-        let open Qmp in
+    | true  ->
         let path = "/machine/peripheral" in
-        let qmp_cmd = Command (None, Qom_list (path)) in
-        let parse_qmp_message = function
-          | Success (None, Qom usbs) -> List.map (fun p -> p.name) usbs
-          | _ -> debug "Get unexpected result after sending Qmp message: %s" (string_of_message qmp_cmd); []
-        in
-        let qmp_cmd_result = qmp_write_and_read domid qmp_cmd in
-        match qmp_cmd_result with
-        | Some qmp_message -> debug "qom_list message result %s" (string_of_message qmp_message);
-          parse_qmp_message qmp_message
-        | None -> debug "Fail to get result after sending Qmp message: %s" (string_of_message qmp_cmd); [])
+        qmp_send_cmd domid Qmp.(Qom_list path)
+        |> ( function
+           | Qmp.(Qom usbs) -> List.map (fun p -> p.Qmp.name) usbs
+           | other ->
+               debug "%s unexpected QMP result for domid %d Qom_list"
+               __LOC__ domid;
```

Comment:
This is an interesting approach, at first I thought the loglines would be too long, but if your editor wraps long lines I think it should still be readable, right?